File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        if len(nums) == 1:
            return 1
        
        nums = list(set(nums))
        nums.sort()
        conslist = []
        currlist = [nums[0]]
        for num in nums[1:]:
            if num - currlist[-1] == 1:
                currlist.append(num)
            else:
                conslist.append(currlist)
                currlist = [num]
        conslist.append(currlist)
        return max([len(x) for x in conslist])
            

            
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const numMap = {}
        for (let num of nums) {
            if (!numMap[num]) {
                numMap[num] = 0;
            }
            numMap[num] += 1;
        }
        const result = Object.entries(numMap)
            .sort((a, b) => b[1] - a[1])
            .slice(0, k)
            .map(entry => Number(entry[0]));
        return result;
    }
}
